Lab1_Operationalizing_Pytorch_with_Mlflow/src/bert_classifier/train.py

```diff
@@ -175,125 +179,100 @@ def prepare_data(self):
         )
 
         self.train_data_loader = self.create_data_loader(
-            self.df_train, self.tokenizer, self.MAX_LEN, self.BATCH_SIZE
+            self.df_train, self.tokenizer, self.max_len, self.batch_size
         )
         self.val_data_loader = self.create_data_loader(
-            self.df_val, self.tokenizer, self.MAX_LEN, self.BATCH_SIZE
+            self.df_val, self.tokenizer, self.max_len, self.batch_size
         )
         self.test_data_loader = self.create_data_loader(
-            self.df_test, self.tokenizer, self.MAX_LEN, self.BATCH_SIZE
+            self.df_test, self.tokenizer, self.max_len, self.batch_size
         )
 
-    def setOptimizer(self):
+    def set_optimizer(self, model):
         """
         Sets the optimizer and scheduler functions
         """
         self.optimizer = AdamW(model.parameters(), lr=1e-3, correct_bias=False)
-        self.total_steps = len(self.train_data_loader) * self.EPOCHS
+        self.total_steps = len(self.train_data_loader) * self.epochs
 
         self.scheduler = get_linear_schedule_with_warmup(
             self.optimizer, num_warmup_steps=0, num_training_steps=self.total_steps
         )
 
         self.loss_fn = nn.CrossEntropyLoss().to(self.device)
 
-    def startTraining(self, model):
-        """
-        Initialzes the Traning step with the model initialized
-
-        :param model: Instance of the NewsClassifier class
-        """
-        history = defaultdict(list)
-        best_accuracy = 0
-
-        for epoch in range(self.EPOCHS):
-
-            print(f"Epoch {epoch + 1}/{self.EPOCHS}")
-
-            train_acc, train_loss = self.train_epoch(model)
-
-            print(f"Train loss {train_loss} accuracy {train_acc}")
-
-            val_acc, val_loss = self.eval_model(model, self.val_data_loader)
-            print(f"Val   loss {val_loss} accuracy {val_acc}")
-
-            history["train_acc"].append(train_acc)
-            history["train_loss"].append(train_loss)
-            history["val_acc"].append(val_acc)
-            history["val_loss"].append(val_loss)
-
-            if val_acc > best_accuracy:
-                torch.save(model.state_dict(), "best_model_state.bin")
-                best_accuracy = val_acc
+    def save_model(self, model):
+        with monit.section('Save model'):
+            mlflow.pytorch.log_model(model, "bert-model",
+                                     registered_model_name="BertModel",
+                                     extra_files=["class_mapping.json", "bert_base_uncased_vocab.txt"])
```

Comment:
@anayden So I will add the python files to extra_files and you can access them?

Reply:
yeah, I think this shoudl work

Reply:
added that